libraries/MotorController/motorControls.cpp

```diff
@@ -0,0 +1,146 @@
+// Easy Driver - Stepper Motor Driver
+// ROB-12779
+//***********************************************************************************************************************************
+// Commanding the length of a step requires pulling MS1/MS2 to differing states
+// (MS1/MS2) : full step (0,0), half step (1,0), 1/4 step (0,1), and 1/8 step (1,1 : default).
+
+// Configuration for taking a full length step
+void MotorControls::fullStep()
+{
+	// MS1 LOW MS2 LOW = full step
+    digitalWrite(MS1, LOW);
+    digitalWrite(MS2, LOW);
+    digitalWrite(STEP, HIGH);  // Actually move the motor
+	delay(1);  // Microsecond units Minimum 1uS PIN HIGH
+	digitalWrite(STEP, LOW); // We are done moving lets go to the bar
+}
+
+// Configuration for taking a half length step
+void MotorControls::halfStep()
+{
+	// MS1 HIGH MS2 LOW = half step
+    digitalWrite(MS1, HIGH);
+    digitalWrite(MS2, LOW);
+    digitalWrite(STEP, HIGH);  // Actually move the motor
+	delay(1);  // Microsecond units Minimum 1uS PIN HIGH
+	digitalWrite(STEP, LOW); // We are done moving lets go to the bar
+}
+
+// Configuration for taking a quarter length step
+void MotorControls::quarterStep()
+{
+	// MS1 LOW MS2 HIGH = quarter step
+    digitalWrite(MS1, LOW);
+    digitalWrite(MS2, HIGH);
+    digitalWrite(STEP, HIGH);  // Actually move the motor
+	delay(1);  // Microsecond units Minimum 1uS PIN HIGH
+	digitalWrite(STEP, LOW); // We are done moving lets go to the bar
+}
+
+// Configuration for taking a quarter length step
+void MotorControls::eighthStep()
+{
+	// MS1 HIGH MS2 HIGH = eighth step
+    digitalWrite(MS1, HIGH);
+    digitalWrite(MS2, HIGH);
+    digitalWrite(STEP, HIGH);  // Actually move the motor
+	delay(1);  // Microsecond units Minimum 1uS PIN HIGH
+	digitalWrite(STEP, LOW); // We are done moving lets go to the bar
+}
+//***********************************************************************************************************************************
+
+// Enable controller to command motor operation
+void MotorControls::turnMotorOn()
+{
+    digitalWrite(SLP, HIGH);
+    digitalWrite(RST, HIGH);
+}
+
+// Pass in number of steps to desired position and the direction
+// param [in] moveSteps 		  - Number of steps between current position and desired new position
+// param [in] direction 		  - Enum indicates direction (reverse: 0/forward: 1/unknown: 2/total: 3) of travel
+// param [in] stepSize (Optional) - Number defining size of step (default: 0)
+void MotorControls::moveMotor(int moveSteps, enum direction, int stepSize = 0)
+{
+	// Prepare the Motor for movement we need direction and number of steps
+	digitalWrite(DIR,direction);  // set the direction high is left low is r5ight
+		
+	// Lets move the specified number of steps	
+	for(int i = 0; i < moveSteps; ++i)
+	{
+		fullStep();				// Simplicity allow only full step option. Half, quarter and eighth also possible
+		
```

Comment:
This is just lazy code, you took the time to create the step size functions, finish this, all the hard work is done.  You can use an enum and and switch case case here to select the step size, you can still default to a full step

Reply:
Added the switch case and created the enum within MotorControllerConstants.h